### ISSUE TITLE
Output for AUX

### ISSUE THREAD

**Issue by Yasunari15** [T-F-S/tcolorbox#47]
Guten Morgen! I have trouble compiling with tcolorbox.sty. I often use the high functioning editor named TeX2img, and it can detect the necessary number of compiling. I found the compiling on TeX2img became slow including tcolorbox because of `\write\@auxout{\string\providecommand\string\tcolorbox@label[2]{}}%` written in the line 965.

There are many tools that keep compiling until AUX file stops changing, so this isn't  just a problem which only appears on TeX2img. I wonder it should output an AUX file in only necessary situations.

Lastly, I recommend you TeX2img because it is so useful and I think it is one of the indispensable tools while using TeX. I itemize some of the function of TeX2img.

* Changing your source, PDF, PS or EPS to EPS, PDF, SVG, SVGZ, EMF, PNG, JPEG, GIF, TIFF and BMP
* these output files includes the original sources and those can be restored
* own preambles, engines and scripts are available

You can download from these pages!

* [Mac](https://github.com/doraTeX/TeX2img/releases/tag/v.2.3.0)
* [Windows](https://github.com/abenori/TeX2img/releases/tag/2.1.0)

**Comment by T-F-S** [comment]
The AUX file(s) is written with every compilation including possibly a large number of lines. 

I have some problems in seeing the point what harm that specific line does... (?)

**Comment by Yasunari15** [comment]
> The AUX file(s) is written with every compilation including possibly a large number of lines.

I’ve heard many build tools ignore a AUX file's `\relax` because this is the default of LaTeX as you said.

> I have some problems in seeing the point what harm that specific line does... (?)

I'm sorry for my obscure explanation. I think the harmful point of `\write\@auxout{\string\providecommand\string\tcolorbox@label[2]{}}%` is to write an unnecessary line in AUX files.

**Comment by T-F-S** [comment]
The line in question is part of the page tracking mechanism of `tcolorbox` which I implemented three years ago: 13-Nov-2015 in my protocol. 

I tried to remember the specifc reason for that line. Clearly, it is a safety feature to prevent a compilation error in case that `\tcolorbox@label` is not (yet) defined. But, now, I was not able to reconstruct such a case. There are two possibilities:

- There is no tricky case any more, because of the current implementation.
- I overlooked the tricky case.

I will deactivate that line in hope for the best. If anything troublesome comes up, I will activate it again.
